File: social-service/src/main/java/com/eirsteir/coffeewithme/social/domain/friendship/FriendshipId.java
package com.eirsteir.coffeewithme.social.domain.friendship;

import com.eirsteir.coffeewithme.social.domain.user.User;
import java.io.Serializable;
import javax.persistence.Embeddable;
import javax.persistence.ManyToOne;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@Embeddable
public class FriendshipId implements Serializable {

  private static final long serialVersionUID = 3966996285633364115L;

  @ManyToOne private User requester;

  @ManyToOne private User addressee;
}

File: social-service/src/main/kotlin/com/eirsteir/coffeewithme/social/domain/friendship/FriendshipId.kt
package com.eirsteir.coffeewithme.social.domain.friendship

import com.eirsteir.coffeewithme.social.domain.user.User
import lombok.Builder
import lombok.Data
import java.io.Serializable
import javax.persistence.Embeddable
import javax.persistence.ManyToOne

@Embeddable
data class FriendshipId(
    @ManyToOne
    val requester: User? = null,
    @ManyToOne
    val addressee: User? = null
) : Serializable {

    companion object {
        private const val serialVersionUID = 3966996285633364115L
    }
}